### PR TITLE
freeoffice: init at 966

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

FreeOffice is a freeware office suite consisting of:

- A word processor: TextMaker
- A spreadsheet: PlanMaker
- A presentation package: Presenter

FreeOffice is a stripped down version of SoftMaker Office, which is not
freeware.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/office/freeoffice/desktop_items.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  planmaker = makeDesktopItem {
+    name = "${pname}-planmaker";
+    desktopName = "FreeOffice PlanMaker";
+    icon = "${pname}-pml.png";
+    categories = "Application;Office;SpreadSheet;";
+    exec = "${pname}-planmaker %F";
+    mimeType = "application/x-pmd;application/x-pmdx;application/x-pmv;application/excel;application/x-excel;application/x-ms-excel;application/x-msexcel;application/x-sylk;application/x-xls;application/xls;application/vnd.ms-excel;application/vnd.stardivision.calc;application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;application/vnd.openxmlformats-officedocument.spreadsheetml.template;application/vnd.ms-excel.sheet.macroenabled.12;application/vnd.openxmlformats-officedocument.spreadsheetml.template;application/vnd.ms-excel.template.macroEnabled.12;application/x-dif;text/spreadsheet;text/csv;application/x-prn;application/vnd.ms-excel.sheet.binary.macroenabled.12";
+    extraEntries = ''
+      TryExec=${pname}-planmaker
+      StartupWMClass=tm
```
Comment (jtojnar) [inline pkgs/applications/office/freeoffice/desktop_items.nix]: Do all the apps really have the same WM_CLASS?

Comment (danieldk) [reply pkgs/applications/office/freeoffice/desktop_items.nix]: Good catch! This was a copy & paste error. Fixed now.

**Comment by aaronjanse** [comment]
Do the wm classes have to be two-letter combos? Is there a reason to not name them, for example, `planmaker`, `presentations`, and `textmaker`?

Or, maybe prefix the class names with `freeoffice` (in case `presentations`, or one of the two-letter combos, is already used)?

**Comment by jtojnar** [comment]
They need to match the `WM_CLASS` property of a window. You can find it out by running `nix run -f '<nixpkgs>' xorg.xprop -c xprop WM_CLASS` and clicking the window in question.

**Comment by aaronjanse** [comment]
> They need to match the WM_CLASS property of a window

Yep, [you're right](https://askubuntu.com/a/809353).

**Comment by aaronjanse** [comment]
Also, I've played around with this expression, and in terms of functionality, it LGTM.
I'd love to have this in the next NixOS release, if possible.

**Comment by danieldk** [comment]
> > They need to match the WM_CLASS property of a window
> 
> Yep, [you're right](https://askubuntu.com/a/809353).

Yep, unfortunately, they have decided to use these two-letter names. Oh well, as long as everyone else isn't ;).

**Comment by danieldk** [comment]
@jtojnar I hope I have addressed all your comments. Would be nice to have this in 19.09.

**Comment by danieldk** [comment]
It turns out that it is easy to add Softmaker Office as well, while sharing a large part of the derivation. I will create a new PR adding both FreeOffice and Softmaker Office.

**Pull request closed without merging**